src/hivpy/demographics.py

```diff
@@ -193,3 +229,27 @@ def initialise_age(self, count):
             age_distribution = ContinuousAgeDistribution.select_model(self.params['inc_cat'])
 
         return age_distribution.gen_ages(count)
+
+    def determine_deaths(self, population_data: pd.DataFrame) -> pd.Series:
+        """Get which individuals die in a time step, as a boolean Series."""
+        all_died = pd.Series([False] * len(population_data))
+        for sex in SexType:
+            for (low_age, high_age), rate in self.params["death_rates"][sex].items():
+                # Find everyone in the relevant age range
+                index = selector(population_data, sex=(operator.eq, sex),
+                                 age=(between, (low_age, high_age)))
+                group = population_data[index]
+                # Probability of dying, assuming time step of 3 months
+                prob_of_death = 1 - exp(-rate / 4)
+                died = np.random.choice([True, False], size=len(group),
+                                        p=[prob_of_death, 1 - prob_of_death])
+                # Mark those who died from this group
+                all_died |= pd.Series(died, index=group.index)
+        return population_data.date_of_death.isnull() & all_died
+
+        # rates = np.ones(len(population_data)) * self.params["death_rate"]
+        # Assuming time step of 3 months
+        # probs = np.exp(-rates / 4)
+        probs = 1 - np.exp(-self.params["death_rate"] / 4)
+        return (population_data.date_of_death.isnull()
+                & np.random.choice([True, False], size=len(population_data), p=[probs, 1-probs]))
```

Comment:
I didn't notice this earlier but this code is after the return statement and never executed, and appears to be an older version of this function, so we should probably cut this!